partner_contact_sale_info_propagation/models/res_partner.py

```diff
@@ -18,67 +15,43 @@ def _check_propagation_allowed(self):
         )
 
     def write(self, vals):
-        """Propagate Salesperson and Sales Channel change in the partner to the
-        child contacts."""
+        """
+        Propagate Salesperson and Sales Channel change in the partner to the
+        child contacts, and ensure team_id is inherited from parent_id.
+        """
         if not self._check_propagation_allowed():
             return super().write(vals)
         for record in self:
             if "user_id" in vals:
                 childs = record.mapped("child_ids").filtered(
-                    lambda r: not r.user_id or r.user_id == record.user_id
+                    lambda r, user_id=record.user_id: not r.user_id
+                    or r.user_id == user_id
```

Comment:
The variable should be named after what represents, and it's a record, not an ID (the only exception here is the fields, which suffixes `_id` for noting that they are m2o records):

```suggestion
                    lambda r, user=record.user_id: not r.user_id
                    or r.user_id == user
```

Do the same for the rest of the occurrences.

Reply:
done, review it again! thanks!